pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -268,6 +294,27 @@ func (os *OpenStack) DetachDisk(instanceID, volumeID string) error {
 	return nil
 }
 
+// ExpandVolume expands the size of specific cinder volume (in GiB)
+func (os *OpenStack) ExpandVolume(volumeID string, newSize int) error {
+	volume, err := os.getVolume(volumeID)
+	if err != nil {
+		return err
+	}
+	if volume.Status != VolumeAvailableStatus {
+		// cinder volume can not be expanded if its status is not available
+		return fmt.Errorf("volume status is not available")
+	}
+
+	volumes, err := os.volumeService("")
+	if err != nil {
+		return err
+	}
+
+	newSizeGb := newSize / 1024 / 1024 / 1024
```

Comment:
In stead of dividing by 1024, can we use `volume.RoundUpSize(expansionSize, 1024*1024*1024)`  ? Also as I commented below - most volume types after resizing also return new size. We should stick to that convention.

Reply:
yeah, will do